docs/system_overview/data_objects.md

```diff
@@ -0,0 +1,343 @@
+
+![Unlink](../images/unlink_01.drawio.png)
+
+For input, unlink requires only an existing data object which is not locked.
+
+The data object must exist in order for unlink to be *possible* in any case.
+
+### Rename
+
+The Rename operation manipulates the logical path of a data object by changing it to a new path.
+
+![Rename](../images/rename_01.drawio.png)
+
+Rename requires 2 inputs:
+
+1. An existing data object which is not locked
+2. Path to a destination data object
+
+If the destination path exists in the logical namespace, rename is *allowed* only if the path refers to a data object and the client has specified a forced overwrite. Overwriting a collection with a data object is not allowed.
+
+## `R_DATA_MAIN`
+
+Data objects and replicas - as with any iRODS entity - are defined in the iRODS Catalog. The information is stored in the `R_DATA_MAIN` table. Each row in `R_DATA_MAIN` describes a single replica of a data object. Some system metadata is identical for each replica, such as data ID, associated collection, and name -- these are defining features of the data object which maps to the replica. Other system metadata varies between replicas, such as host resource and replica number -- these are defining features of the individual replicas.
+
+The following are all of the columns representing the system metadata in the table as well as their meanings:
+
+ - `data_id`: Identifier unique to the data object that maps to this replica. This column will contain the same value for all replicas of a given data object.
+ - `coll_id`: Identifier unique to the collection associated with the data object that maps to this replica. This column will contain the same value for all replicas of a given data object. The identifiers are defined in `R_COLL_MAIN` which contains all of the information about collections in a given zone.
+ - `data_name`: Logical Name of the data object that maps to this replica. This column will contain the same value for all replicas of a given data object. This value will be unique for all data objects with an identical `coll_id`.
+ - `data_repl_num`: Identifier unique to this replica. This value does not have any special meaning in the context of the iRODS system other than being able to identify a replica independently of its host resource.
+ - `data_version`: This column is not touched by iRODS anymore. Its purpose is lost to history.
+ - `data_type_name`:
+ - `data_size`: Size of the replica in bytes. This may vary between replicas depending on the state of the data and may or may not reflect the state of the data in storage. `ifsck` can be used to detect inconsistencies.
+ - `resc_group_name`: This column is not touched by iRODS anymore. It refers to the legacy resource groups feature. This should contain the value `EMPTY_RESC_GROUP_NAME` in order to discourage legacy queries.
```

Comment:
"This column is no longer used by iRODS. The value `EMPTY_RESC_GROUP_NAME` is used to discourage legacy queries."

Reply:
Done